### DISCUSSION TITLE
Can I use a model in FileNameGeneratorInterface?

### DISCUSSION THREAD

**Discussion by klaaskox** [Laravel-Backpack/community-forum#843] [Q&A (Help)] [answered]
Hi,

I can use a model's ID when generating the filename for an image that is uploaded so I can store it as 971.png for example?

Thanks!

**Comment by karandatwani92** [comment, marked as answer]
Hey @klaaskox 

The file name is generated before saving. There is no model ID before saving.

For other cases, You can [create a custom FileNameGenerator](https://backpackforlaravel.com/docs/6.x/crud-uploaders#naming-files-when-using-uploaders-1) and use it for naming files.

If you want to name it model ID:
- You can do it in alternative ways of course, for example, you can write your upload function that first saves the model, gets the ID, then saves the file with the ID, and updates the model value. The guide to override the store method is [here](https://backpackforlaravel.com/docs/6.x/crud-operation-create#override-the-store-method).
- Alternatively,  It's recommended to use [media library uploaders](https://backpackforlaravel.com/docs/6.x/crud-uploaders#uploaders-for-spatie-medialibrary) to link uploaded media to a model.
- Some would put `$nextId  = DB::table('user')->max('id') + 1;` in FileNameGenerator. (It's Bad practice.)

**Reply by klaaskox** [reply]
Thanks a lot @karandatwani92! I didn't know the filename was generated before processing the model, but now it makes sense.
